tree/ntuple/src/RPageStorageFile.cxx

```diff
@@ -305,6 +317,15 @@ ROOT::Internal::RPageSourceFile::CreateFromAnchor(const RNTuple &anchor, const R
    return pageSource;
 }
 
+std::unique_ptr<ROOT::Internal::RPageSourceFile>
+ROOT::Internal::RPageSourceFile::OpenWithDifferentAnchor(const RNTuple &anchor, const ROOT::RNTupleReadOptions &options)
+{
+   auto pageSource = std::make_unique<RPageSourceFile>("", fFile->Clone(), options);
+   pageSource->fAnchor = anchor;
+   pageSource->fNTupleName = pageSource->fDescriptorBuilder.GetDescriptor().GetName();
```

Comment:
How is this initialized? Above you are passing `""` for `ntupleName`, is the header already loaded somewhere?

Reply:
BY "initialized" you mean Attached? At the moment it needs to be done manually, like when you create a new PageSource. This is done in 2 places in the [full PR](https://github.com/root-project/root/pull/19153): in `ReadAttributeSet`, by the created Reader, and in `RNTupleMerger::OpenAttributeSource`

Reply:
I see. It's indeed possible that that line is useless (I still need to verify), but isn't this the case also for `CreateFromAnchor` then?

Reply:
https://github.com/root-project/root/pull/20017